### PR TITLE
ADD: focal loss

### PR DESCRIPTION
**Pull request by benleetownsend**

### CODE REVIEW

File: finetune/nn/target_blocks.py [hunk 1]
```diff
@@ -170,6 +170,8 @@ def classifier(hidden, targets, n_targets, config, train=False, reuse=None, **kw
             clf_losses = tf.nn.softmax_cross_entropy_with_logits_v2(
                 logits=clf_logits, labels=tf.stop_gradient(targets)
             )
+            if config.focal_loss:
+                clf_losses *= (1 - tf.reduce_sum(tf.nn.softmax(clf_logits) * targets, -1)) ** config.focal_loss_gamma
```
Comment (madisonmay) [inline finetune/nn/target_blocks.py]: In the original focal loss paper they also use an alpha of 0.25 for a gamma of 2.  Might be worth adding in

**Comment by madisonmay** [comment]
Think this needs a quick rebase / cherry-pick
